Automattic/newspack-block-theme
# feat: add drawer menu pattern; reorganise overlays

### All Submissions:






### Changes proposed in this Pull Request:

This PR does a few things:

1. The SASS overlay files have been reorganised and they are in their own folder
2. I added a generic `--position--full-width` class to the `overlay-contents`
3. You can also control its width `--width--x-small`, `--width--small`, `--width--large`, `--width--x-large`
4. I added the open class to the contents container so we can automatically trigger the opening animation without requiring other custom CSS
5. There's a new pattern for a Drawer Menu. It comes with a new template part (Drawer Content) so users can add any content they want to this menu.

Screencast: (note: this is slightly outdated and doesn't use `--position`)
https://github.com/Automattic/newspack-block-theme/assets/177929/fa193950-3bc0-4c7b-805a-d889a6ec5f9e

### How to test the changes in this Pull Request:

1. Switch to this PR
2. Add the Drawer Menu pattern to a page
3. Test if it works
4. In the editor, select the Drawer Content and change the overlay-contents style (left | right | full-width)
5. Change also its width. (check UTILITY_CLASSES.md for documentation)
6. Test the front-end
7. Test all the other overlays (mobile menu, search, comments) if they still work as expected

### Other information:
